test-suite/src/function/ifnull.rs

```diff
@@ -0,0 +1,74 @@
+use crate::*;
+
+test_case!(ifnull, async move {
+    use chrono::{NaiveDate, NaiveDateTime, NaiveTime};
+    use gluesql_core::{executor::Payload, prelude::Value::*};
+    use rust_decimal::Decimal;
+    let test_cases = vec![
+        (
+            r#"CREATE TABLE SingleItem (id integer null, int8 int(8) null, dec decimal null, 
+                                        dt date null, mystring Text null,
+                                        mybool Boolean null, myfloat float null,
+                                        mytime time null, mytimestamp timestamp null)"#,
+            Payload::Create,
+        ),
+        (
+            r#"INSERT INTO SingleItem VALUES (0, 1, 2, "2022-05-23", "this is a string", true, 3.15,
+                          "01:02:03", "1970-01-01 00:00:00 -00:00")"#,
+            Payload::Insert(1),
+        ),
+        (
+            r#"INSERT INTO SingleItem VALUES (null, null, null, null, null, null, null, null, null)"#,
+            Payload::Insert(1),
+        ),
+        (
+            r#"SELECT ifnull(id, 1) as ID, ifnull(int8, 2) as int8, ifnull(dec, 3) 
+            FROM SingleItem where id is not null"#,
```

Comment:
One also important role of integration tests is providing usage guide to users.
In this kind of SQL examples, both using all-uppercased or all-lowercased for SQL keywords look ok.
However, mixing uppercase and lowercase letters would be confusing for users.
```
SELECT IFNULL(id, 1) AS ID, IFNULL(INT8, 2) AS int8, IFNULL(dec, 3)
FROM SingleItem WHERE id IS NOT NULL
```
or
```
select ifnull(id, 1) as ID, ifnull(int8, 2) as int8, ifnull(dec, 3)
from SingleItem where id is not null
```

Reply:
yes, I have a bad habit of typing it in lower case..